src/App.tsx

```diff
@@ -2,24 +1,44 @@
+import React, { useState, useCallback } from 'react';
+import { Pagination } from './components/Pagination';
 import './App.css';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+  const handlePageChange = useCallback((page: number) => {
+    setCurrentPage(page);
+  }, []);
+
+  const handlePerPageChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(Number(e.target.value));
+    setCurrentPage(1); // на 1 ст. при зміні к-ті на ст.
+  };
+
+  const startIndex = (currentPage - 1) * perPage;
+  const paginatedItems = items.slice(startIndex, startIndex + perPage);
```

Comment:
The calculation for `startIndex` and `paginatedItems` can be encapsulated into a custom hook or a utility function to separate the pagination logic from the component and make it reusable.